hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/dynamicconfig/WanReplicationConfigTransformer.java

```diff
@@ -0,0 +1,275 @@
+    }
+
+
+    @Nullable
+    public WanConsumerConfigHolder toHolder(@Nullable WanConsumerConfig config) {
+        if (config == null) {
+            return null;
+        }
+        return new WanConsumerConfigHolder(
+                config.isPersistWanReplicatedData(),
+                config.getClassName(),
+                serializationService.toData(config.getImplementation()),
+                toHolderProperties(config.getProperties()));
+    }
+
+    @Nonnull
+    public WanCustomPublisherConfigHolder toHolder(@Nonnull WanCustomPublisherConfig config) {
+        checkNotNull(config, "WAN custom publisher config must be provided");
+        return new WanCustomPublisherConfigHolder(
+                config.getPublisherId(),
+                config.getClassName(),
+                serializationService.toData(config.getImplementation()),
+                toHolderProperties(config.getProperties()));
+    }
+
+    @Nonnull
+    public Map<String, Data> toHolderProperties(@Nullable Map<String, Comparable> m) {
+        if (m == null) {
+            return Collections.emptyMap();
+        }
+
+        Map<String, Data> safeMapping = new HashMap<>();
+        for (Map.Entry<String, Comparable> e : m.entrySet()) {
+            safeMapping.put(e.getKey(), serializationService.toData(e.getValue()));
+        }
+        return safeMapping;
+    }
+
+    @Nonnull
+    public WanBatchPublisherConfigHolder toHolder(@Nonnull WanBatchPublisherConfig config) {
+        checkNotNull(config, "WAN batch publisher config must be provided");
+        return new WanBatchPublisherConfigHolder(
+                config.getPublisherId(),
+                config.getClassName(),
+                serializationService.toData(config.getImplementation()),
+                toHolderProperties(config.getProperties()),
+                config.getClusterName(),
+                config.isSnapshotEnabled(),
+                config.getInitialPublisherState().getId(),
+                config.getQueueCapacity(),
+                config.getBatchSize(),
+                config.getBatchMaxDelayMillis(),
+                config.getResponseTimeoutMillis(),
+                config.getQueueFullBehavior().getId(),
+                config.getAcknowledgeType().getId(),
+                config.getDiscoveryPeriodSeconds(),
+                config.getMaxTargetEndpoints(),
+                config.getMaxConcurrentInvocations(),
+                config.isUseEndpointPrivateAddress(),
+                config.getIdleMinParkNs(),
+                config.getIdleMaxParkNs(),
+                config.getTargetEndpoints(),
+                config.getAwsConfig(),
+                config.getGcpConfig(),
+                config.getAzureConfig(),
+                config.getKubernetesConfig(),
+                config.getEurekaConfig(),
+                serializationService.toData(config.getDiscoveryConfig()),
+                config.getSyncConfig().getConsistencyCheckStrategy().getId(),
```

Comment:
The sync config looks to have always been a basic wrapper around an enum. @vbekiaris any thoughts on that?

Reply:
@srknzl I have added `WanSyncConfig` codec -- as you say, easier to make it a codec just in case someone wants to add bits in the future. See https://github.com/hazelcast/hazelcast/pull/25118/commits/94b06d695e047de0b8a046cfef1c6507be19c6a3

Reply:
A codec sounds better 👍